examples/qtensor/qtensor.morpho

```diff
@@ -0,0 +1,152 @@
+import meshtools
+import optimize
+import plot
+import povray
+
+var refine_adaptively = false
+// Define physical parameters
+
+var rho = 1.3 // Deep in the nematic phase
+var EA = 3 // Anchoring strength
+var K = 1.0 // Bending modulus
+
+var a2 = (1-rho)
+var a4 = (1+rho)/rho^2
+
+// Import disk mesh
+
+var m = Mesh("disk.mesh")
+var m = refinemesh(m) // Refining for a better result
+
+// Select boundary
+var bnd = Selection(m, boundary=true)
+bnd.addgrade(0) // add point elements
+
+// Create director field
+
+/* Since a 2D Q-tensor for a uniaxial nematic is symmetric and 
+traceless, there are only two independent components, Qxx and Qxy*/
+
+var q_tensor = Field(m, fn(x,y,z) Matrix([0.01*random(1), 0.01*random(1)]))
+
+// Define bulk free energy
+
+fn landau(x, q) {
+  var qt = q.norm()
+  var qt2=qt*qt
+  return a2*qt2 + a4*qt2*qt2
+}
+
+// Define anchoring energy at the boundary
+fn anchoring(x, q) {
+  var t = tangent()
+  var wxx = t[0]*t[0]-0.5
+  var wxy = t[0]*t[1]
+  return (q[0]-wxx)^2+(q[1]-wxy)^2
+}
+
+var bulk = AreaIntegral(landau, q_tensor)
+var anchor = LineIntegral(anchoring, q_tensor)
+var elastic = GradSq(q_tensor)
+
+var problem = OptimizationProblem(m)
+problem.addenergy(bulk)
+problem.addenergy(elastic, prefactor = K)
+problem.addenergy(anchor, selection=bnd, prefactor = EA)
+
+var opt = FieldOptimizer(problem, q_tensor)
+// Minimize the free energy!
+opt.linesearch(500)
+
+
+// Refine the mesh based on the elastic energy density/
+// This for loop can be commented out 
+if (refine_adaptively){
+  for (i in 1..4){
+  
+    // Select elements that have a large contribution to the energy
+    var en = elastic.integrand(q_tensor)
+    var mean = en.sum()/en.count()
+    var srefine = Selection(m) // Start with an empty selection
+    for (id in 0...en.count()) if (en[0,id]>1.5*mean) srefine[2,id]=true // Identify high energy elements
```

Comment:
What does "high energy" mean here (I see that is elements that are over %150 of the mean but you should say that)